### PR TITLE
Add custom headers and expected status code ranges support.

### PR DESCRIPTION
**Pull request by ttlcgl**
…ell SDK.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by maddieclayton** [comment]
@tatlicioglu Has this been through cmdlet review? If not please fill out one here: https://github.com/Azure/azure-powershell-cmdlet-review-pr.  If so, please link in the description.

**Comment by maddieclayton** [comment]
@tatlicioglu Ping

**Comment by cormacpayne** [comment]
@tatlicioglu closing this PR due to lack of activity. Please feel free to re-open this PR when you are able to address the above comments.

File: src/ResourceManager/TrafficManager/Commands.TrafficManager2/Commands.TrafficManager.csproj [hunk 1]
```diff
@@ -96,10 +90,9 @@
       <Link>AzureRM.TrafficManager.psd1</Link>
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </None>
+    <None Include="app.config" />
```
Comment (cormacpayne) [inline src/ResourceManager/TrafficManager/Commands.TrafficManager2/Commands.TrafficManager.csproj]: @allencal please remove this reference to `app.config`

File: src/ResourceManager/TrafficManager/Commands.TrafficManager2/Commands.TrafficManager.csproj [hunk 2]
```diff
@@ -111,6 +104,21 @@
       <SubType>Designer</SubType>
     </EmbeddedResource>
   </ItemGroup>
+  <ItemGroup>
+    <Reference Include="Microsoft.Azure.Management.TrafficManager, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.TrafficManager.2.4.0-preview\lib\net452\Microsoft.Azure.Management.TrafficManager.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (cormacpayne) [inline src/ResourceManager/TrafficManager/Commands.TrafficManager2/Commands.TrafficManager.csproj]: @allencal you can remove the references to `Microsoft.Rest.ClientRuntime`, `Microsoft.Rest.ClientRuntime.Azure`, `Newtonsoft.Json` and `System.Runtime` added in this file

File: src/ResourceManager/TrafficManager/Commands.TrafficManager2/packages.config [hunk 3]
```diff
@@ -1,4 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Azure.Management.TrafficManager" version="2.3.0-preview" targetFramework="net452" />
-</packages>
+  <package id="Microsoft.Azure.Management.TrafficManager" version="2.4.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.4.0" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/TrafficManager/Commands.TrafficManager2/packages.config]: @tatlicioglu same comment about removing the references to `Microsoft.Rest.ClientRuntime`, `Microsoft.Rest.ClientRuntime.Azure` and `Newtonsoft.Json` from this file

File: src/ResourceManager/TrafficManager/Commands.TrafficManager2/Commands.TrafficManager.csproj [hunk 4]
```diff
@@ -111,6 +104,21 @@
       <SubType>Designer</SubType>
     </EmbeddedResource>
   </ItemGroup>
+  <ItemGroup>
+    <Reference Include="Microsoft.Azure.Management.TrafficManager, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (cormacpayne) [inline src/ResourceManager/TrafficManager/Commands.TrafficManager2/Commands.TrafficManager.csproj]: @allencal you'll need to update the version of the SDK used in the [.NET Core project](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/TrafficManager/Commands.TrafficManager2/Commands.TrafficManager.Netcore.csproj#L35) for the Travis build to pass

**Comment by cormacpayne** [comment]
Closing this PR due to lack of activity. @tatlicioglu @allencal feel free to re-open this PR (or create a new one) when this is being actively worked on.

**Pull request closed without merging**